example-plugins/abnormal_lab_task_notification/protocols/abnormal_lab_protocol.py

```diff
@@ -0,0 +1,74 @@
+from canvas_sdk.effects import Effect
+from canvas_sdk.effects.task import AddTask, TaskStatus
+from canvas_sdk.events import EventType
+from canvas_sdk.protocols import BaseProtocol
+from canvas_sdk.v1.data.lab import LabReport
+from logger import log
+
+
+class AbnormalLabProtocol(BaseProtocol):
+    """
+    A protocol that monitors lab reports and creates task notifications
+    for abnormal lab values to ensure prompt review.
+
+    Triggers on: LAB_REPORT_CREATED events
+    Effects: Creates tasks for abnormal lab values
+    """
+
+    RESPONDS_TO = EventType.Name(EventType.LAB_REPORT_CREATED)
+
+    def compute(self) -> list[Effect]:
+        """
+        This method gets called when a LAB_REPORT_CREATED event is fired.
+        It checks for abnormal lab values and creates tasks for them.
+        """
+        # Get the lab report ID from the event target
+        lab_report_id = self.event.target.id
+
+        try:
+            # Get the lab report instance with filters applied
+            lab_report = LabReport.objects.filter(
+                id=lab_report_id,
+                for_test_only=False,
+                junked=False,
+                patient__isnull=False
+            ).first()
+            
+            if not lab_report:
+                return []
+            
+            patient_id = lab_report.patient.id
+
+            # Check all lab values for abnormal flags
+            abnormal_values = []
+            for lab_value in lab_report.values.all():
+                # Check if the lab value has an abnormal flag (handle None case)
+                abnormal_flag = getattr(lab_value, 'abnormal_flag', None) or ""
+                if abnormal_flag.strip():
+                    abnormal_values.append(lab_value)
```

Comment:
i'm really nitpicking now, but to make this really succinct, i think we could do:
```
from django.db.models.functions import Trim

abnormal_values = list(report.values.annotate(trimmed_abnormal_flag=Trim('abnormal_flag')).exclude(trimmed_abnormal_flag='').values_list("trimmed_abnormal_flag", flat=True))
```
but depends on whether that import is allowed.